code/modules/scrap/oldificator.dm

```diff
@@ -0,0 +1,250 @@
+		brokenmodule.make_old()
+		qdel(src)
+	..()
+
+/obj/item/clothing/suit/space/make_old()
+	if(prob(75))
+		var/datum/breach/B = new()
+		breaches += B
+		B.class = pick(25,50,75)
+		B.damtype = pick(BRUTE, BURN)
+		B.update_descriptor()
+		B.holder = src
+	..()
+
+/obj/item/clothing/make_old()
+	if(prob(50))
+		slowdown += pick(0.5, 0.5, 1, 1.5)
+	if(prob(75))
+		armor["melee"] = armor["melee"] / 2
+		armor["bullet"] = armor["bullet"] / 2
+		armor["laser"] = armor["laser"] / 2
+		armor["energy"] = armor["energy"] / 2
+		armor["bomb"] = armor["bomb"] / 2
+		armor["bio"] = armor["bio"] / 2
+		armor["rad"] = armor["rad"] / 2
+	if(prob(25))
+		heat_protection = 0
+	if(prob(25))
+		cold_protection = 0
+	if(prob(35))
+		contaminate()
+	if(prob(75))
+		add_blood()
+	if(prob(50))
+		equip_delay += rand(0, 2 SECONDS)
+	..()
+
+/obj/item/weapon/aiModule/broken
+	name = "\improper broken core AI module"
+	desc = "broken Core AI Module: 'Reconfigures the AI's core laws.'"
+
+/obj/machinery/broken/Initialize()
+	..()
+	explosion(loc, 1, 2, 3, 3)
+	return INITIALIZE_HINT_QDEL
+
+/obj/machinery/broken/Destroy()
+	contents.Cut()
+	return ..()
+
+/obj/item/weapon/aiModule/broken/transmitInstructions(mob/living/silicon/ai/target, mob/sender)
+	..()
+	IonStorm(0)
+	explosion(sender.loc, 1, 1, 1, 3)
+	sender.drop_from_inventory(src)
+	qdel(src)
+
+/obj/item/weapon/dnainjector/make_old()
+	if(prob(75))
+		name = "DNA-Injector (unknown)"
+		desc = pick("1mm0r74l17y 53rum", "1ncr3d1bl3 73l3p47y hNlk", "5up3rhum4n m16h7")
+		value = 0xFFF
+	if(prob(75))
+		block = pick(MONKEYBLOCK, HALLUCINATIONBLOCK, DEAFBLOCK, BLINDBLOCK, NERVOUSBLOCK, TWITCHBLOCK, CLUMSYBLOCK, COUGHBLOCK, HEADACHEBLOCK, GLASSESBLOCK)
+	..()
+
+/obj/item/clothing/glasses/hud/make_old()
+	if(prob(75) && !istype(src, /obj/item/clothing/glasses/hud/broken))
+		var/obj/item/clothing/glasses/hud/broken/brokenhud = new /obj/item/clothing/glasses/hud/broken
+		brokenhud.name = src.name
+		brokenhud.desc = src.desc
+		brokenhud.icon = src.icon
+		brokenhud.icon_state = src.icon_state
+		brokenhud.item_state = src.item_state
+		brokenhud.make_old()
+		qdel(src)
+	..()
+
+/obj/item/clothing/glasses/make_old()
+	..()
+	if(prob(75))
+		vision_flags = 0
+	if(prob(75))
+		darkness_view = -1
+
+/obj/item/device/lighting/glowstick/make_old()
+	..()
+	if(prob(75))
+		fuel = rand(0, fuel)
+
+/obj/item/device/lighting/toggleable/make_old()
+	..()
+	if(prob(75))
+		brightness_on = brightness_on / 2
+
+/obj/machinery/floodlight/make_old()
+	..()
+	if(prob(75))
+		brightness_on = brightness_on / 2
+
+/obj/machinery/make_old()
+	..()
+	if(prob(60))
+		stat |= BROKEN
+	if(prob(60))
+		emagged = TRUE
+
+/obj/machinery/vending/make_old()
+	..()
+	if(prob(60))
+		seconds_electrified = -1
+	if(prob(60))
+		shut_up = 0
+	if(prob(60))
+		shoot_inventory = 1
+	if(prob(75))
+		var/del_count = rand(0,product_records.len)
+		for(var/i = 1 to del_count)
+			var/removed_item = pick(contents)
+			contents -= removed_item
```

Comment:
i've heavily modified vendors, this part should be reviewed

they don't contain items anymore until those items are vended